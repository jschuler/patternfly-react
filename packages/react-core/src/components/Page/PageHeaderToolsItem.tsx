import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Page/page';
import { css } from '@patternfly/react-styles';
import { PageHeaderToolsBreakpointMod } from './PageHeaderTools';
import { formatBreakpointMods } from '../../helpers/util';

interface InjectedPageHeaderToolsItemProps {
  selectedClass: string;
}

export interface PageHeaderToolsItemProps extends React.HTMLProps<HTMLDivElement> {
  /** Content rendered in page header tools item. If you need the selected class, use the render prop instead */
  children?: React.ReactNode;
  /** If you need the selected class, use this render prop instead of children */
  render?: (props: InjectedPageHeaderToolsItemProps) => React.ReactNode;
  /** Additional classes added to the page header tools item. */
  className?: string;
  /** An array of breakpoint modifiers to control visibility, e.g. breakpointMods={[{ modifier: 'hidden' }, { modifier: 'visible', breakpoint: 'md' }]} */
  breakpointMods?: PageHeaderToolsBreakpointMod[];
}

export const PageHeaderToolsItem: React.FunctionComponent<PageHeaderToolsItemProps> = ({
  children,
  render,
  className,
  breakpointMods
}: PageHeaderToolsItemProps) => (
  <div
    className={css(
      styles.pageHeaderToolsItem,
      breakpointMods && formatBreakpointMods(breakpointMods, styles),
      className
    )}
  >
    {children}
    {render && render({ selectedClass: styles.modifiers.selected })}
  </div>
);
