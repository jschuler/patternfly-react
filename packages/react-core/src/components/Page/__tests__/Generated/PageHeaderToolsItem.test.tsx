/**
 * This test was generated
 */
import * as React from 'react';
import { shallow } from 'enzyme';
import { PageHeaderToolsItem } from '../../PageHeaderToolsItem';
// any missing imports can usually be resolved by adding them here
import {} from '../..';

it('PageHeaderToolsItem should match snapshot (auto-generated)', () => {
  const view = shallow(
    <PageHeaderToolsItem
      children={<div>ReactNode</div>}
      visibleOnLg={true}
      hiddenOnLg={true}
      hiddenOnSm={true}
      isSelected={true}
    />
  );
  expect(view).toMatchSnapshot();
});
