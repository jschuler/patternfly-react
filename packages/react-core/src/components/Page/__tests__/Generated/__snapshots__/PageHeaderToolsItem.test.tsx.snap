// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeaderToolsItem should match snapshot (auto-generated) 1`] = `
<div
  className="pf-c-page__header-tools-item string"
>
  <div>
    ReactNode
  </div>
</div>
`;
